cdap-gateway/src/test/java/co/cask/cdap/gateway/router/AuthServerAnnounceTest.java

```diff
@@ -0,0 +1,172 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.gateway.router;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.conf.SConfiguration;
+import co.cask.cdap.common.guice.ConfigModule;
+import co.cask.cdap.common.guice.DiscoveryRuntimeModule;
+import co.cask.cdap.common.guice.IOModule;
+import co.cask.cdap.internal.guava.reflect.TypeToken;
+import co.cask.cdap.security.auth.AccessTokenTransformer;
+import co.cask.cdap.security.guice.SecurityModules;
+import co.cask.cdap.security.server.GrantAccessToken;
+import com.google.common.collect.Maps;
+import com.google.common.io.CharStreams;
+import com.google.common.net.InetAddresses;
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.gson.Gson;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.twill.discovery.DiscoveryService;
+import org.apache.twill.discovery.DiscoveryServiceClient;
+import org.apache.twill.discovery.InMemoryDiscoveryService;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class AuthServerAnnounceTest {
+  private static final String HOSTNAME = "127.0.0.1";
+  private static final int CONNECTION_IDLE_TIMEOUT_SECS = 2;
+  private static final DiscoveryService DISCOVERY_SERVICE = new InMemoryDiscoveryService();
+  private static final String ANNOUNCE_ADDRESS = "vip.cask.co:10000";
+  private static final Gson GSON = new Gson();
+  private static final Type TYPE = new TypeToken<Map<String, List<String>>>() {
+  }.getType();
+
+  @Test
+  public void testEmptyAnnounceAddressConfig() throws Exception {
+    HttpRouterService routerService = createRouterService(null);
+    routerService.startUp();
+    try {
+      DefaultHttpClient client = new DefaultHttpClient();
+      String url = resolveURI(Constants.Router.GATEWAY_DISCOVERY_NAME, "/v3/apps", routerService);
+      HttpGet get = new HttpGet(url);
+      HttpResponse response = client.execute(get);
+      Map<String, List<String>> responseMap =
+        GSON.fromJson(new InputStreamReader(response.getEntity().getContent()), TYPE);
+      Assert.assertEquals(Collections.EMPTY_LIST, responseMap.get("auth_uri"));
+    } finally {
+      routerService.shutDown();
+    }
+  }
+
+  @Test
+  public void testAnnounceAddressConfig() throws Exception {
```

Comment:
create a helper function to reduce duplicated code in `testEmptyAnnounceAddressConfig ` `testAnnounceAddressConfig`?